### ISSUE TITLE
Right click menu still stops images changing.

### ISSUE THREAD

**Issue by Wil-B** [TheQwertiest/foo_spider_monkey_panel#50]
So, for example, if the menu is showing then on track change the text changes but the image doesn't: so its a mismatch.

Is this an unavoidable consequence of using utils.GetAlbumArtAsyncV2 & gdi.LoadImageAsyncV2 or fixable?

**Comment by TheQwertiest** [comment]
> Is this an unavoidable consequence of using utils.GetAlbumArtAsyncV2 & gdi.LoadImageAsyncV2?

Regretfully, yes, it is unavoidable, since it's a restriction by ECMAScript standard. For more info see https://github.com/TheQwertiest/foo_spider_monkey_panel/issues/40#issuecomment-448670047

**Comment by TheQwertiest** [comment]
PPPPS: Updated answer with PS's :D

**Comment by marc2k3** [comment]
Perhaps revert to the old methods?? Getting the album art or a few artist images on track change is an amazingly cheap operation that doesn't need to be async.

**Comment by TheQwertiest** [comment]
> Getting the album art or a few artist images ... is an amazingly cheap operation

It really depends on the amount of images you are loading and the timing of the operation. E.g. my playlist usage (CaTRoX) gets really uncomfortable without async art fetching (esp. for tracks managed by `foo_youtube`). Obviously, it depends on the script and the scenario - YMMV and all that =)

**Comment by marc2k3** [comment]
Absolutely. I was basing my comments on the assumption that Wil was only working on porting his publicly available JSP scripts to SMP. If he's working on any kind of playlist/library viewer, that would be another matter entirely.
